src/util/utility-types.ts

```diff
@@ -4,3 +4,23 @@
  * Required since React.ComponentProps doesn't think it's a JSXElementConstructor/ReactElement. https://github.com/FB-PLP/traject/pull/11929/files#r948531375
  */
 export type ExtractProps<T> = T extends React.ComponentType<infer P> ? P : T;
+
+/**
+ * Given two types, only allow one of them to be valid
+ */
```

Comment:
Could you add a blurb about if the types are missing, will the ts error message will say it'll prefer the first type?